src/Npgsql/Internal/NpgsqlConnector.Auth.cs

```diff
@@ -190,13 +190,15 @@ internal void AuthenticateSASLSha256Plus(ref string mechanism, ref string cbindF
             throw new NpgsqlException("Server offered SCRAM-SHA-256-PLUS authentication over a non-SSL connection");
 
         var sslStream = (SslStream)_stream;
-        if (sslStream.RemoteCertificate is null)
+        // The remote certificate we get here is actually X509Certificate2
```

Comment:
Is this a 100% safe assumption, i.e. is there a chance this will throw InvalidCastException?

Should we instead check for an X509Certificate that isn't X509Certificate2, and fall back to the current logic if that's the case (i.e. wrap RemoteCertificate in X509Certificate2 and dispose it later)?

Reply:
Currently (and at least since .NET 5, I'm too lazy to check earlier versions) the backing field for that property is [X509Certificate2](https://github.com/dotnet/runtime/blob/fa494644343910cb9931e5ca0bd6857ea74e1c07/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs#L57), so I would say it's extremely unlikely to ever throw `InvalidCastException`. Of course it's not impossible for the runtime to change the type, but given how averse they are to introduce changes like that (and it's not like there is any class what suits better), I think we should be fine.

But I'm OK with making it like it was before, it's still a very cold path, so no one would particularly care.

Reply:
Yeah, we shouldn't rely on the type of a private backing field - if it's all the same, better to fall back to the previous previous (agree that it's super cold and has no impact on anything).